lib/origin/selector.rb

```diff
@@ -18,7 +18,10 @@ class Selector < Smash
     def merge!(other)
       other.each_pair do |key, value|
         if value.is_a?(Hash) && self[key.to_s].is_a?(Hash)
-          value = self[key.to_s].merge(value)
+          value = self[key.to_s].merge(value) do |_key, oldval, newval|
+            return newval if !multi_variable?(_key)
```

Comment:
In general, instead of doing 'if' followed by '!', you can use 'unless'

In any case, I wouldn't use the 'return' word here. You could use the ternary operator:

```ruby
multi_variable?(_key) ? (old_val + new_val).uniq : new_val
```